### PR TITLE
MuonAlignment ASAN bug fix and DB output update

### LINKED ISSUE #43: Deprecated legacy modules in `Alignment/MuonAlignment*`
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ae46bb/19774/llvm-analysis/

```Deprecated API	inherits from edm::EDProducer,edm::EDFilter,edm::EDAnalyzer, or edm::OutputModule	/Alignment/MuonAlignmentAlgorithms/plugins/MuonAlignmentPreFilter.cc	MuonAlignmentPreFilter	30	1	View Report
Deprecated API	inherits from edm::EDProducer,edm::EDFilter,edm::EDAnalyzer, or edm::OutputModule	/Alignment/MuonAlignmentAlgorithms/plugins/CSCOverlapsBeamSplashCut.cc	CSCOverlapsBeamSplashCut	40	1	View Report
Deprecated API	inherits from edm::EDProducer,edm::EDFilter,edm::EDAnalyzer, or edm::OutputModule	/Alignment/MuonAlignment/plugins/MuonGeometrySVGTemplate.cc	MuonGeometrySVGTemplate	46	1	View Report
```

### LINKED ISSUE #36053: [ASAN] heap-buffer-overflow in MuonAlignmentInputDB::newAlignableMuon
Log file: https://cmssdt.cern.ch/SDT/cgi-bin/logreader/slc7_amd64_gcc10/CMSSW_12_2_ASAN_X_2021-11-08-2300/unitTestLogs/Alignment/MuonAlignment#/27

Error message:
```
==15449==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x604000e3f900 at pc 0x2b4ebd8148a0 bp 0x7fff48447cf0 sp 0x7fff48447ce8
READ of size 8 at 0x604000e3f900 thread T0
    #0 0x2b4ebd81489f in MuonAlignmentInputDB::newAlignableMuon() const (.../lib/slc7_amd64_gcc10/libAlignmentMuonAlignment.so+0x11789f)
    #1 0x2b4ebd7e2f07 in MuonAlignment::MuonAlignment(edm::EventSetup const&, MuonAlignmentInputMethod const&) (.../lib/slc7_amd64_gcc10/libAlignmentMuonAlignment.so+0xe5f07)
    #2 0x2b4e9b5ce876 in MuonGeometryDBConverter::analyze(edm::Event const&, edm::EventSetup const&) (.../lib/slc7_amd64_gcc10/pluginMuonGeometryDBConverter.so+0x3b876)
    #3 0x2b4e8b40e36f in edm::one::EDAnalyzerBase::doEvent(edm::EventTransitionInfo const&, edm::ActivityRegistry*, edm::ModuleCallingContext const*) (.../lib/slc7_amd64_gcc10/libFWCoreFramework.so+0x8be36f)
    #4 0x2b4e8b3b2bd2 in edm::WorkerT<edm::one::EDAnalyzerBase>::implDo(edm::EventTransitionInfo const&, edm::ModuleCallingContext const*) (.../lib/slc7_amd64_gcc10/libFWCoreFramework.so+0x862bd2)
    #5 0x2b4e8b09d634 in decltype ({parm#1}()) edm::convertException::wrap<edm::Worker::runModule<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >(edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::TransitionInfoType const&, edm::StreamID, edm::ParentContext const&, edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::Context const*)::{lambda()#1}>(edm::Worker::runModule<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >(edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::TransitionInfoType const&, edm::StreamID, edm::ParentContext const&, edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::Context const*)::{lambda()#1}) (.../lib/slc7_amd64_gcc10/libFWCoreFramework.so+0x54d634)
    #6 0x2b4e8b09de7a in std::__exception_ptr::exception_ptr edm::Worker::runModuleAfterAsyncPrefetch<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >(std::__exception_ptr::exception_ptr const*, edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::TransitionInfoType const&, edm::StreamID, edm::ParentContext const&, edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1>::Context const*) (.../lib/slc7_amd64_gcc10/libFWCoreFramework.so+0x54de7a)
    #7 0x2b4e8b0ac914 in void edm::SerialTaskQueueChain::actionToRun<edm::Worker::RunModuleTask<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >::execute()::{lambda()#1}&>(edm::Worker::RunModuleTask<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >::execute()::{lambda()#1}&) (.../lib/slc7_amd64_gcc10/libFWCoreFramework.so+0x55c914)
    #8 0x2b4e8b0ad674 in edm::SerialTaskQueue::QueuedTask<edm::SerialTaskQueueChain::push<edm::Worker::RunModuleTask<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >::execute()::{lambda()#1}&>(tbb::detail::d1::task_group&, edm::Worker::RunModuleTask<edm::OccurrenceTraits<edm::EventPrincipal, (edm::BranchActionType)1> >::execute()::{lambda()#1}&)::{lambda()#1}>::execute() (.../lib/slc7_amd64_gcc10/libFWCoreFramework.so+0x55d674)
    #9 0x2b4e8bda5344 in tbb::detail::d1::function_task<edm::SerialTaskQueue::spawn(edm::SerialTaskQueue::TaskBase&)::{lambda()#1}>::execute(tbb::detail::d1::execution_data&) (.../lib/slc7_amd64_gcc10/libFWCoreConcurrency.so+0xa344)
    #10 0x2b4e8d3ecb8b in tbb::detail::d1::task* tbb::detail::r1::task_dispatcher::local_wait_for_all<false, tbb::detail::r1::external_waiter>(tbb::detail::d1::task*, tbb::detail::r1::external_waiter&) /data/cmsbld/jenkins/workspace/auto-builds/CMSSW_12_1_0_pre5-slc7_amd64_gcc10/build/CMSSW_12_1_0_pre5-build/BUILD/slc7_amd64_gcc10/external/tbb/v2021.4.0-544b7b2b990e0660abde669a014ec673/tbb-v2021.4.0/src/tbb/task_dispatcher.h:322
    #11 0x2b4e8d3ecb8b in tbb::detail::d1::task* tbb::detail::r1::task_dispatcher::local_wait_for_all<tbb::detail::r1::external_waiter>(tbb::detail::d1::task*, tbb::detail::r1::external_waiter&) /data/cmsbld/jenkins/workspace/auto-builds/CMSSW_12_1_0_pre5-slc7_amd64_gcc10/build/CMSSW_12_1_0_pre5-build/BUILD/slc7_amd64_gcc10/external/tbb/v2021.4.0-544b7b2b990e0660abde669a014ec673/tbb-v2021.4.0/src/tbb/task_dispatcher.h:463
    #12 0x2b4e8d3ecb8b in tbb::detail::r1::task_dispatcher::execute_and_wait(tbb::detail::d1::task*, tbb::detail::d1::wait_context&, tbb::detail::d1::task_group_context&) /data/cmsbld/jenkins/workspace/auto-builds/CMSSW_12_1_0_pre5-slc7_amd64_gcc10/build/CMSSW_12_1_0_pre5-build/BUILD/slc7_amd64_gcc10/external/tbb/v2021.4.0-544b7b2b990e0660abde669a014ec673/tbb-v2021.4.0/src/tbb/task_dispatcher.cpp:168
    #13 0x2b4e8adff5bb in edm::EventProcessor::processLumis(std::shared_ptr<void> const&) (.../lib/slc7_amd64_gcc10/libFWCoreFramework.so+0x2af5bb)
    #14 0x2b4e8ae2f082 in edm::EventProcessor::runToCompletion() (.../lib/slc7_amd64_gcc10/libFWCoreFramework.so+0x2df082)
    #15 0x40e961 in tbb::detail::d1::task_arena_function<main::{lambda()#1}::operator()() const::{lambda()#1}, void>::operator()() const (.../bin/slc7_amd64_gcc10/cmsRun+0x40e961)
    #16 0x2b4e8d3db15a in tbb::detail::r1::task_arena_impl::execute(tbb::detail::d1::task_arena_base&, tbb::detail::d1::delegate_base&) /data/cmsbld/jenkins/workspace/auto-builds/CMSSW_12_1_0_pre5-slc7_amd64_gcc10/build/CMSSW_12_1_0_pre5-build/BUILD/slc7_amd64_gcc10/external/tbb/v2021.4.0-544b7b2b990e0660abde669a014ec673/tbb-v2021.4.0/src/tbb/arena.cpp:698
    #17 0x411e56 in main::{lambda()#1}::operator()() const (.../bin/slc7_amd64_gcc10/cmsRun+0x411e56)
    #18 0x40c605 in main (.../bin/slc7_amd64_gcc10/cmsRun+0x40c605)
    #19 0x2b4e8e219554 in __libc_start_main (/lib64/libc.so.6+0x22554)
    #20 0x40c948  (.../bin/slc7_amd64_gcc10/cmsRun+0x40c948)

0x604000e3f900 is located 0 bytes to the right of 48-byte region [0x604000e3f8d0,0x604000e3f900)
allocated by thread T0 here:
    #0 0x2b4e8a2333d7 in operator new(unsigned long) ../../../../libsanitizer/asan/asan_new_delete.cpp:99
    #1 0x2b4e963d6a82 in __gnu_cxx::new_allocator<double>::allocate(unsigned long, void const*) .../external/gcc/10.3.0/include/c++/10.3.0/ext/new_allocator.h:115
    #2 0x2b4e963d6a82 in std::allocator_traits<std::allocator<double> >::allocate(std::allocator<double>&, unsigned long) .../external/gcc/10.3.0/include/c++/10.3.0/bits/alloc_traits.h:460
    #3 0x2b4e963d6a82 in std::_Vector_base<double, std::allocator<double> >::_M_allocate(unsigned long) .../external/gcc/10.3.0/include/c++/10.3.0/bits/stl_vector.h:346
    #4 0x2b4e963d6a82 in std::_Vector_base<double, std::allocator<double> >::_M_create_storage(unsigned long) .../external/gcc/10.3.0/include/c++/10.3.0/bits/stl_vector.h:361
    #5 0x2b4e963d6a82 in std::_Vector_base<double, std::allocator<double> >::_Vector_base(unsigned long, std::allocator<double> const&) .../external/gcc/10.3.0/include/c++/10.3.0/bits/stl_vector.h:305
    #6 0x2b4e963d6a82 in std::vector<double, std::allocator<double> >::vector(unsigned long, std::allocator<double> const&) .../external/gcc/10.3.0/include/c++/10.3.0/bits/stl_vector.h:511
    #7 0x2b4e963d6a82 in CLHEP::HepSymMatrix::HepSymMatrix(int, int) /data/cmsbld/jenkins/workspace/auto-builds/CMSSW_12_1_0_pre3-slc7_amd64_gcc10/build/CMSSW_12_1_0_pre3-build/BUILD/slc7_amd64_gcc10/external/clhep/2.4.5.1-e54099ca360a82ea7047a04132b31b4b/clhep-2.4.5.1/Matrix/src/SymMatrix.cc:73
```

### PR DESCRIPTION
**Pull request by hyunyong**
#### PR description:
Resolve https://github.com/cms-sw/cmssw/issues/36053
Resolve https://github.com/cms-AlCaDB/AlCaTools/issues/43
DB output service migration for MuonAlignment (https://github.com/cms-AlCaDB/AlCaTools/issues/28)

### CODE REVIEW

File: Alignment/MuonAlignment/plugins/MuonMisalignedProducer.cc [hunk 1]
```diff
@@ -84,7 +84,7 @@ MuonMisalignedProducer::MuonMisalignedProducer(const edm::ParameterSet& p)
       esTokenGEM_(esConsumes(edm::ESInputTag("", "idealForMuonMisalignedProducer"))) {}
 
 //__________________________________________________________________________________________________
-MuonMisalignedProducer::~MuonMisalignedProducer() = default;
+MuonMisalignedProducer::~MuonMisalignedProducer() {}
```
Comment (tvami) [inline Alignment/MuonAlignment/plugins/MuonMisalignedProducer.cc:87]: I think the `= default;` is the recommended way of doing this

File: Alignment/MuonAlignment/src/MuonAlignment.cc [hunk 2]
```diff
@@ -284,8 +284,8 @@ void MuonAlignment::saveCSCSurveyToDB(void) {
     throw cms::Exception("NotAvailable") << "PoolDBOutputService not available";
 
   // Get alignments and errors
-  Alignments cscAlignments{};
-  SurveyErrors cscSurveyErrors{};
+  Alignments* cscAlignments = new Alignments();
```
Comment (tvami) [inline Alignment/MuonAlignment/src/MuonAlignment.cc:287]: here the original was the preferred solution

File: Alignment/MuonAlignment/plugins/MuonMisalignedProducer.cc (hunk 1 above)
Comment (hyunyong) [reply Alignment/MuonAlignment/plugins/MuonMisalignedProducer.cc:87]: The code-format tool changed this. do you want to keep ```= default;```?

**Comment by tvami** [comment]
Hi @hyunyong I wrote to you privately, you did many things in the PR from which some are correct and some are undoing the work of @mmusich about the DB output module changes

File: Alignment/MuonAlignment/plugins/MuonMisalignedProducer.cc (hunk 1 above)
Comment (tvami) [reply Alignment/MuonAlignment/plugins/MuonMisalignedProducer.cc:87]: Interesting, @makortel @smuzaffar I thought the  `default` was the preferred option

Comment (hyunyong) [reply Alignment/MuonAlignment/plugins/MuonMisalignedProducer.cc:87]: > Hi @hyunyong I wrote to you privately, you did many things in the PR from which some are correct and some are undoing the work of @mmusich about the DB output module changes

Oh, ok. 
I copied the module from an old version.
I understood now what happened.
I will close this PR and make a new one.

Comment (makortel) [reply Alignment/MuonAlignment/plugins/MuonMisalignedProducer.cc:87]: It is, and I'm a bit surprised that code-checks would change away from `= default`. We have plenty of those in `.cc` files.

Comment (hyunyong) [reply Alignment/MuonAlignment/plugins/MuonMisalignedProducer.cc:87]: > It is, and I'm a bit surprised that code-checks would change away from `= default`. We have plenty of those in `.cc` files.

no, it was my mistake.

**Pull request closed without merging**